bitcoin/src/blockdata/witness.rs

```diff
@@ -249,6 +249,14 @@ impl Witness {
         witness
     }
 
+    /// Creates a witness required to do a key path spend of a P2TR output.
+    pub fn p2tr_key_spend(signature: &taproot::Signature) -> Witness {
+        let mut witness = Witness::new();
+        let serialized = signature.to_vec();
```

Comment:
:open_mouth: we're missing `serialize()` method, which would avoid allocation.

Reply:
ooo yeah, thanks.